invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -309,112 +316,114 @@ def load(self, ignore=None):
+        """Template method design pattern for loading entries."""
+        ignore = ignore or set()
+        self.pre_load(identity, ignore=ignore)
+        for data in self.iterate(ignore=ignore):
+            self.create_record(data, delay=delay)
+        return self.loaded()
 
-    @property
-    def covered_ids(self):
-        """Just the id of the vocabulary covered by this entry as a list."""
-        return [self._id]
+    def create_record(self, data, delay=False):
+        """Create the record."""
+        if delay:
+            create_vocabulary_record.delay(self.service_str, data)
+        else:  # mostly for tests
+            create_vocabulary_record(self.service_str, data)
 
-    def create_vocabulary_type(self, identity):
-        """Create the vocabulary type."""
-        pid_type = self._entry['pid-type']
-        current_service.create_type(identity, self._id, pid_type)
 
-    def load(self, identity, ignore=None, delay=None):
-        """Load the data file if self._id not in ignore.
+class GenericVocabularyEntry(VocabularyEntry):
+    """Vocabulary fixture with single data-file."""
 
-        Return the loaded id as 1 item list.
-        """
-        ignore = ignore or set()
+    def __init__(self, directory, id_, entry):
+        """Constructor."""
+        super().__init__("vocabulary_service", directory, id_, entry)
 
+    # Template methods
+    def pre_load(self, identity, ignore):
+        """Actions taken before iteratively creating records."""
         if self._id not in ignore:
-            self.create_vocabulary_type(identity)
+            pid_type = self._entry['pid-type']
+            current_service.create_type(identity, self._id, pid_type)
 
-            for record in self:
-                record['type'] = self._id
-                if delay:
-                    create_vocabulary_record.delay(self._service_str, record)
-                else:  # mostly for tests
-                    create_vocabulary_record(self._service_str, record)
-            return [self._id]
+    def iterate(self, ignore):
+        """Iterate over dicts of file content."""
+        if self._id not in ignore:
+            filepath = self._dir / self._entry["data-file"]
+            for data in create_iterator(filepath):
+                data['type'] = self._id
+                yield data
 
-        return []
+    def loaded(self):
+        """Vocabularies actually loaded."""
+        return [self._id]
 
+    # Other interface methods
+    @property
+    def covered_ids(self):
+        """Just the id of the vocabulary covered by this entry as a list."""
+        return [self._id]
 
-class SubjectsVocabularyEntry:
-    """Vocabulary fixture for subjects vocabulary."""
 
-    def __init__(self, directory, id_, entry):
-        """Constructor."""
-        self._dir = directory
-        self._parent_id = id_
-        self._entry = entry
+class SpecificVocabularyEntryWithSchemes(VocabularyEntry):
```

Comment:
What about `VocabularyEntryWithSchemes` to make it a bit shorter... I think is more or less clear that when using schemes they cannot go to the generic vocab db/index?

Reply:
:+1: I was ahead of myself